### PR TITLE
[fix bug 933375] Only show current and future votes on the first vote…

### PR DESCRIPTION
**Pull request by tsmrachel**
… listing page

https://bugzilla.mozilla.org/show_bug.cgi?id=933375

### CODE REVIEW

File: remo/voting/views.py [hunk 1]
```diff
@@ -34,22 +34,19 @@ def list_votings(request):
...
-    past_polls_paginator = Paginator(past_polls_query, settings.ITEMS_PER_PAGE)
-    past_polls_page = request.GET.get('page', 1)
-
-    try:
-        past_polls = past_polls_paginator.page(past_polls_page)
-    except PageNotAnInteger:
-        past_polls = past_polls_paginator.page(1)
-    except EmptyPage:
-        past_polls = past_polls_paginator.page(past_polls_paginator.num_pages)
+    show_all_past = request.GET.get('show_all_past','false')
```
Comment (Ryuno-Ki) [inline remo/voting/views.py]: Hm, maybe casting that call to a Boolean in order to minimise the risk of XSS...

File: remo/voting/views.py [hunk 2]
```diff
@@ -34,22 +34,19 @@ def list_votings(request):
...
 
+    if show_all_past == 'true':
+        past_polls = polls.filter(end__lt=now())
+    else :
```
Comment (Ryuno-Ki) [inline remo/voting/views.py]: Doesn't your linter yell at you here about the space before :?

File: remo/voting/views.py [hunk 3]
```diff
@@ -34,22 +34,19 @@ def list_votings(request):
...
+        past_polls = polls.filter(end__lt=now())[:10]
```
Comment (Ryuno-Ki) [inline remo/voting/views.py]: Personally I feel like you should pass the `10` as `per_page` variable to the template, so to keep both values in sync. Means, make it a variable/symbolic constant.

File: remo/voting/views.py [hunk 4]
```diff
@@ -34,22 +34,19 @@ def list_votings(request):
...
+    
     return render(request, 'list_votings.jinja',
                   {'user': user,
+                   'show_all_past' : show_all_past,
```
Comment (Ryuno-Ki) [inline remo/voting/views.py]: Again, space before the :

File: remo/voting/views.py (hunk 3 above)
Comment (tsmrachel) [inline remo/voting/views.py]: Sorry @Ryuno-Ki , i didn't quite understand what you meant. The previous assignee for this bug stated that after consultation with @akatsoulas , it should show 10 of the latest previous votes, and show all previous votes after clicking on a link. Why is there a need to pass in 10 as a per_page variable?

**Comment by akatsoulas** [comment]
Thank you @tsmrachel  for this pull request. I will review this as soon as possible. In the meantime, here are the linting errors from Travis:

remo/voting/views.py:1:1: F401 'django.conf.settings' imported but unused
remo/voting/views.py:4:1: F401 'django.core.paginator.EmptyPage' imported but unused
remo/voting/views.py:4:1: F401 'django.core.paginator.PageNotAnInteger' imported but unused
remo/voting/views.py:4:1: F401 'django.core.paginator.Paginator' imported but unused
remo/voting/views.py:41:52: E231 missing whitespace after ','
remo/voting/views.py:48:1: W293 blank line contains whitespace

File: remo/voting/views.py [hunk 5]
```diff
@@ -23,6 +21,7 @@ def list_votings(request):
     """List votings view."""
     user = request.user
     polls = Poll.objects.all()
+    PAST_POLLS_PER_PAGE = 10
```
Comment (vprasanth) [inline remo/voting/views.py]: Seems kind of weird this breaks the lower case convention followed else where in the code.

**Comment by MichaelKohler** [comment]
That was the wrong tab, will have a look at this PR in a few minutes, but that comment was not meant for this one here. Sorry!

**Pull request closed without merging**